testing/events_test.go

```diff
@@ -0,0 +1,223 @@
+package ibctesting_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+)
+
+func TestParsePacketsFromEvents(t *testing.T) {
+	tests := []struct {
+		name            string
+		events          []abci.Event
+		expectedPackets []channeltypes.Packet
+		expectedError   string
+	}{
+		{
+			name:          "fail: no events",
+			events:        []abci.Event{},
+			expectedError: "ibctesting.ParsePacketsFromEvents: acknowledgement event attribute not found",
+		},
+		{
+			name: "fail: events without packet",
+			events: []abci.Event{
+				{
+					Type: "xxx",
+				},
+				{
+					Type: "yyy",
+				},
+			},
+			expectedError: "ibctesting.ParsePacketsFromEvents: acknowledgement event attribute not found",
+		},
+		{
+			name: "fail: event packet with invalid AttributeKeySequence",
+			events: []abci.Event{
+				{
+					Type: channeltypes.EventTypeSendPacket,
+					Attributes: []abci.EventAttribute{
+						{
+							Key:   channeltypes.AttributeKeySequence,
+							Value: "x",
+						},
+					},
+				},
+			},
+			expectedError: "ibctesting.ParsePacketsFromEvents: strconv.ParseUint: parsing \"x\": invalid syntax",
+		},
+		{
+			name: "fail: event packet with invalid AttributeKeyTimeoutHeight",
+			events: []abci.Event{
+				{
+					Type: channeltypes.EventTypeSendPacket,
+					Attributes: []abci.EventAttribute{
+						{
+							Key:   channeltypes.AttributeKeyTimeoutHeight,
+							Value: "x",
+						},
+					},
+				},
+			},
+			expectedError: "ibctesting.ParsePacketsFromEvents: expected height string format: {revision}-{height}. Got: x: invalid height",
+		},
+		{
+			name: "fail: event packet with invalid AttributeKeyTimeoutTimestamp",
+			events: []abci.Event{
+				{
+					Type: channeltypes.EventTypeSendPacket,
+					Attributes: []abci.EventAttribute{
+						{
+							Key:   channeltypes.AttributeKeyTimeoutTimestamp,
+							Value: "x",
+						},
+					},
+				},
+			},
+			expectedError: "ibctesting.ParsePacketsFromEvents: strconv.ParseUint: parsing \"x\": invalid syntax",
+		},
+		{
+			name: "ok: events with packets",
```

Comment:
We usually have the success case the first on of the slice.

```suggestion
			name: "success",
```